pkg/networkmanager/openvpn.jsx

```diff
@@ -0,0 +1,112 @@
+import { FormGroup } from '@patternfly/react-core/dist/esm/components/Form/index.js';
+import { TextInput } from '@patternfly/react-core/dist/esm/components/TextInput/index.js';
+import cockpit from 'cockpit';
+import { ModelContext } from './model-context';
+import { useDialogs } from 'dialogs.jsx';
+
+const _ = cockpit.gettext;
+
+export function OpenVPNDialog({ settings, connection, dev }) {
+    const Dialogs = useDialogs();
+    const idPrefix = "network-openvpn-settings";
+    const model = useContext(ModelContext);
+
+    const [iface, setIface] = useState(settings.connection.interface_name);
+    const [remote, setRemote] = useState("");
+    const [caCertName, setCaCertName] = useState("");
+    const [caCertVal, setCaCertVal] = useState("");
+    const [userCertName, setUserCertName] = useState("");
+    const [userCertVal, setUserCertVal] = useState("");
+    const [userKeyName, setUserKeyName] = useState("");
+    const [userPrivateKey, setUserPrivateKey] = useState("");
+    const [dialogError, setDialogError] = useState("");
+
+    async function onSubmit() {
+        const user = await cockpit.user();
+        const caPath = `${user.home}/.cert/ca-${caCertName}.crt`;
+        const userCertPath = `${user.home}/.cert/cert-${userCertName}`;
+        const userKeyPath = `${user.home}/.cert/key-${userKeyName}.key`;
+        try {
+            await cockpit.script(`mkdir -p ${user.home}/.cert`);
+            await cockpit.script(`touch ${caPath} ${userCertPath} ${userKeyPath}`);
+            await cockpit.file(caPath).replace(caCertVal);
+            await cockpit.file(userCertPath).replace(userCertVal);
+            await cockpit.file(userKeyPath).replace(userPrivateKey);
+        } catch (err) {
+            console.log("an error occured => ", err);
+        }
+
+        function createSettingsObject() {
+            return {
+                ...settings,
+                connection: {
+                    ...settings.connection,
+                    type: 'vpn',
+                },
+                vpn: {
+                    data: {
+                        remote,
+                        ca: caPath,
+                        cert: userCertPath,
+                        key: userKeyPath,
+                        // hardcoding the bellow properties until we have a design for advanced dialog
+                        'cert-pass-flags': '0',
+                        'connection-type': 'tls',
+                        dev: 'tun',
+                        'push-peer-info': 'yes',
+                        'remote-cert-tls': 'server'
+                    },
+                    'service-type': 'org.freedesktop.NetworkManager.openvpn'
+                }
+            };
+        }
+
+        dialogSave({
+            connection,
+            dev,
+            model,
+            settings: createSettingsObject(),
+            onClose: Dialogs.close,
+            setDialogError,
```

Comment:
These 7 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19363-20230921-075022-79cb9fd2-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)